web/src/components/merchant/sign-up/SignUpCard.tsx

```diff
@@ -56,31 +65,123 @@ const StyledButton = styled(Button)`
   font-size: 18px;
   font-weight: bolder;
   text-transform: uppercase;
-
-  margin-top: 30px;
+  margin: 0;
 `;
 
 /**
  * This is the card containing the sign up form for the Merchant app.
  * It is displayed in the Sign Up page.
  */
-const SignUpCard: React.FC = () => (
-  <CardContainer>
-    <GroupBuyMerchantHeader />
-    <Form>
-      <FormRow label="Name" inputType="text" />
-      <FormRow label="Email" inputType="email" />
-      <FormRow label="Password" inputType="password" />
-      <FormRow label="Confirm Password" inputType="password" />
-      <FormRow label="VPA" inputType="email" />
+const SignUpCard: React.FC = () => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState('');
+
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const [vpa, setVpa] = useState('');
+  const [vpaError, setVpaError] = useState('');
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.target.value ? setNameError('') : setNameError(NAME_EMPTY);
+    setName(event.target.value);
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.target.value ? setEmailError('') : setEmailError(EMAIL_EMPTY);
+    setEmail(event.target.value);
+  };
+
+  const handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.target.value
+      ? setPasswordError('')
+      : setPasswordError(PASSWORD_EMPTY);
+    // We don't check that passwords match if user hasn't filled in
+    // 'Confirm Password' yet.
+    !confirmPassword || event.target.value === confirmPassword
+      ? setConfirmPasswordError('')
+      : setConfirmPasswordError(PASSWORDS_DO_NOT_MATCH);
+    setPassword(event.target.value);
+  };
+
+  const handleConfirmPasswordChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    // We don't have to check if 'Confirm Password' is empty. If 'Password' is
+    // is not empty and 'Confirm Password' is empty, we show PASSWORDS_DO_NOT_MATCH.
+    // If both passwords are empty we show PASSWORD_EMPTY.
+    event.target.value === password
+      ? setConfirmPasswordError('')
+      : setConfirmPasswordError(PASSWORDS_DO_NOT_MATCH);
+    setConfirmPassword(event.target.value);
+  };
+
+  const handleVpaChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.target.value ? setVpaError('') : setVpaError(VPA_EMPTY);
+    setVpa(event.target.value);
+  };
+
+  const handleSubmit = () => {
```

Comment:
can you enable the submit button only if there is no error? then no need to have this check?

Reply:
The submit button triggers some checks also, so I think that might be a bit weird.

Reply:
For example, when user's email is already in use by existing account, then the error is only thrown after the user clicks the submit button

Reply:
Okay, actually after thinking about it, it makes sense to enable the submit button only if there is no detectable error at that point. But it requires quite a bit of time to implement. I just feel that this is not a necessary feature and can be implemented later. I think that I should prioritise on other more important features first.

Reply:
ok. I probably left same comment in a different PR

Reply:
I managed to add this feature! Because I am now using a library so adding this feature becomes very easy.